interfaces/builtin/storage_framework.go

```diff
@@ -0,0 +1,153 @@
+
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={RequestName,ReleaseName,GetConnectionCredentials}
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (bind)
+    bus=session
+    name=com.canonical.StorageFramework.Registry,
+
+dbus (bind)
+    bus=session
+    name=com.canonical.StorageFramework.Provider.*,
+`
+
+const storageFrameworkConnectedSlotAppArmor = `
+# Description: Allow clients to access the registry and storage framework services.
+
+#include <abstractions/dbus-session-strict>
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Registry
+    path=/com/canonical/StorageFramework/Registry
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Provider.*
+    path=/provider/*
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const storageFrameworkConnectedPlugAppArmor = `
+# Description: Allow access to the registry and storage framework services.
+
+#include <abstractions/dbus-session-strict>
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Registry
+    path=/com/canonical/StorageFramework/Registry
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Provider.*
+    path=/provider/*
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`
+
+type StorageFrameworkInterface struct{}
+
+func (iface *StorageFrameworkInterface) Name() string {
+	return "storage-framework"
+}
+
+func (iface *StorageFrameworkInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *StorageFrameworkInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		snippet := []byte(storageFrameworkConnectedPlugAppArmor)
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		new := slotAppLabelExpr(slot)
+		snippet = bytes.Replace(snippet, old, new, -1)
+		return snippet, nil
+	}
+	return nil, nil
+}
+
+func (iface *StorageFrameworkInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return []byte(storageFrameworkPermanentSlotAppArmor), nil
+	}
+	return nil, nil
+}
+
+func (iface *StorageFrameworkInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		snippet := []byte(storageFrameworkConnectedSlotAppArmor)
+		old := []byte("###PLUG_SNAP_NAME###")
+		new := []byte(plug.Snap.Name())
+		snippet = bytes.Replace(snippet, old, new, -1)
```

Comment:
###PLUG_SNAP_NAME### isn't currently in storageFrameworkConnectedSlotAppArmor. Was it meant to be used as part of the `path` in this rule:
```
dbus (receive, send)
    bus=session
    interface=com.canonical.StorageFramework.Provider.*
    path=/provider/*
    peer=(label=###PLUG_SECURITY_TAGS###),
```

If so, then I like that idea a lot.